### PR TITLE
Add sandboxing support for OSX

### PR DESCRIPTION
**Pull request by AltGr**

### CODE REVIEW

File: src/state/sandbox_exec.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#!/bin/sh -ue
+
+POL='(version 1)(allow default)(deny network*)(deny file-write*)'
```
Comment (avsm) [inline src/state/sandbox_exec.sh]: is the default behaviour of the Linux one also to deny network connections?  That's a sensible default, but probably shouldn't diverge between operating systems

File: src/state/sandbox_exec.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        add_mounts ro "$OPAM_SWITCH_PREFIX/.opam-switch"
+        add_mounts rw "$PWD"
+        ;;
+    remove)
+        add_mounts rw "$OPAM_SWITCH_PREFIX"
+        add_mounts ro "$OPAM_SWITCH_PREFIX/.opam-switch"
+        [ "X${PWD#$OPAM_SWITCH_PREFIX/.opam-switch}" != "X${PWD}" ] && add_mounts rw "$PWD"
+        ;;
+    *)
+        echo "bwrap.sh: unknown command $COMMAND, must be one of 'build', 'install' or 'remove'" >&2
```
Comment (avsm) [inline src/state/sandbox_exec.sh]: this bwrap.sh could be $0 to be more accurate

File: src/state/sandbox_exec.sh (hunk 1 above)
Comment (AltGr) [reply src/state/sandbox_exec.sh]: yes, it's the same on Linux. There are now built-in mechanisms for package needing to download extra files, so better to enforce them.

File: src/state/sandbox_exec.sh (hunk 2 above)
Comment (AltGr) [reply src/state/sandbox_exec.sh]: Indeed, thanks!

**Comment by avsm** [comment]
working fine for me with a fresh switch -- might need some testing to find out what happens for rc1->rc2 users

File: src/state/sandbox_exec.sh (hunk 1 above)
Comment (avsm) [reply src/state/sandbox_exec.sh]: Thanks, I agree that these are excellent defaults.

**Pull request merged**